pkg/kube/kube_handler_test.go

```diff
@@ -1151,64 +1167,25 @@ func TestValidateAppRepository(t *testing.T) {
 			if tc.expectedError != nil {
 				return
 			}
-			reqs, err := getRequests(appRepo, cli)
+			httpValidator, err := getValidator(appRepo, cli)
 			if (err != nil || tc.expectedReqResolutionError != nil) && !errors.Is(err, tc.expectedReqResolutionError) {
 				t.Fatalf("got: %+v, want: %+v", err, tc.expectedReqResolutionError)
 			}
 			if tc.expectedReqResolutionError != nil {
 				return
 			}
-			reqURLs := []string{}
-			for _, req := range reqs {
-				reqURLs = append(reqURLs, req.URL.String())
-			}
-			if !cmp.Equal(tc.expectedURLs, reqURLs) {
-				t.Errorf("Unexpected URLS: %v", cmp.Diff(tc.expectedURLs, reqURLs))
-			}
-			if tc.expectedHeaders != nil && !cmp.Equal(tc.expectedHeaders, cli.(*httpclient.ClientWithDefaults).DefaultHeaders) {
-				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(tc.expectedHeaders, cli.(*httpclient.ClientWithDefaults).DefaultHeaders))
-			}
```

Comment:
Here - we can't just delete these tests that verified the correct requests and/or headers were set. These tests are still necessary and need to be done somehow as outlined above.

Reply:
I have restored this piece of test code.